dex/src/state.rs

```diff
@@ -1983,13 +2118,37 @@ pub(crate) mod account_parser {
                 None, // To use an open orders authority, explicitly use the
                       // InitOpenOrders instruction.
             )?;
+
+            drop(market);
+            let mut market_state = MarketStateV2::load(market_acc, program_id)?;
```

Comment:
can you move this logic into process_new_order_v4, instead of doing it in the arg parser, because we update the epoch_state there while also updating seq_num, see line 3417 and the 3600

```
order_book_state
            .market_state
            .check_and_update_epoch_cycle(seq_num)?;
```

in which case the logic below can be cleaned up infinite as well. 

Fine to keep this check here tho.

```
            if instruction.tif_offset > market_state.epoch_length {
```